compose.py

```diff
@@ -0,0 +1,305 @@
+                          '--verbose',
+                          action="store_true",
+                          required=False,
+                          help='Enable verbose output.')
+    parser.add_argument(
+        '--build-name',
+        type=str,
+        required=False,
+        help=
+        'Build name. Default is "tritonserver_build".'
+    )
+    parser.add_argument(
+        '--build-dir',
+        type=str,
+        required=True,
+        help=
+        'Build directory. All repo clones and builds will be performed in this directory.'
+    )
+    parser.add_argument(
+        '--version',
+        type=str,
+        required=False,
+        help=
+        'The Triton version. If not specified defaults to the value in the TRITON_VERSION file.'
+    )
+    parser.add_argument(
+        '--container-version',
+        type=str,
+        required=False,
+        help=
+        'The Triton container version to build. If not specified the container version will be chosen automatically based on --version value.'
+    )
+    parser.add_argument(
+        '--upstream-container-version',
+        type=str,
+        required=False,
+        help=
+        'The upstream container version to use for the build. If not specified the upstream container version will be chosen automatically based on --version value.'
+    )
+    parser.add_argument(
+        '--backend',
+        action='append',
+        required=False,
+        help=
+        'Include specified backend in build as <backend-name>. To have multiple backends built, specify --backend <foo> --backend <bar>'
+    )
+    parser.add_argument(
+        '--repoagent',
+        action='append',
+        required=False,
+        help='Include specified repo agent in build as <repoagent-name>')
+    FLAGS = parser.parse_args()
+
+    if FLAGS.build_name is None:
+        FLAGS.build_name = "tritonserver_build"
+    # Determine the versions. Start with Triton version, if --version
+    # is not explicitly specified read from TRITON_VERSION file.
+    if FLAGS.version is None:
+        with open('TRITON_VERSION', "r") as vfile:
+            FLAGS.version = vfile.readline().strip()
+    if FLAGS.container_version is None:
+        if FLAGS.version not in TRITON_VERSION_MAP:
+            fail('container version not known for {}'.format(FLAGS.version))
+        FLAGS.container_version = TRITON_VERSION_MAP[FLAGS.version][0]
+    if FLAGS.upstream_container_version is None:
+        if FLAGS.version not in TRITON_VERSION_MAP:
+            fail('upstream container version not known for {}'.format(
+                FLAGS.version))
+        FLAGS.upstream_container_version = TRITON_VERSION_MAP[
+            FLAGS.version][1]
+
+    dockerfile_name = 'Dockerfile.buildbase'
+    workdir_path="/opt/tritonserver"
+    
+    if FLAGS.backend is None:
+        FLAGS.backend = []
+    if FLAGS.repoagent is None:
+        FLAGS.repoagent = []
+
+    make_dockerfile(FLAGS.build_dir, dockerfile_name)
```

Comment:
This breakout of functions is confusing. I see why you want add_requested_backends and add_requested_repoagents separately... so you can use them in the future cpu-only dockerfile... but other than show I would just have "start_gpu_dockerfile" and "end_gpu_dockerfile" or similar.